app/src/lib/repository-description.ts

```diff
@@ -0,0 +1,42 @@
+import * as Path from 'path'
+import * as Fs from 'fs'
+
+const GitDescriptionPath = '.git/description'
+
+const DefaultGitDescription = "Unnamed repository; edit this file 'description' to name the repository.\n"
+
+/** Get the repository's description from the .git/description file. */
+export function getGitDescription(repositoryPath: string): string {
+  const path = Path.join(repositoryPath, GitDescriptionPath)
+
+  try {
+    const description = Fs.readFileSync(path, 'utf8')
```

Comment:
We should be using the `readFile` asynchronous method here, which means refactoring this method to:

 - the method should return `Promise<string>`
 - because `readFile` is asynchronous the method internally should return a `new Promise<string>(/* ... */)`
 - internally it can just return an empty string when it encounters an error, rather than propagating the error up

An example of this, to reference and steal liberally from:

https://github.com/desktop/desktop/blob/a61a5bdc94ee8237dfff328957cdaee99a9b61e1/app/src/ui/add-repository/gitignores.ts#L48-L66